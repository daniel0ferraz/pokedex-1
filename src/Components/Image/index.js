import React from "react";

export default function Image(props) {
  const { source, alt, size, className, onClick } = props;
  return (
    <div
      onClick={onClick}
      style={{ width: size, display: "flex", alignItems: "center" }}
      className={className}
    >
      <img src={source} alt={alt} style={{ width: "100%", zIndex: 2 }} />
    </div>
  );
}
