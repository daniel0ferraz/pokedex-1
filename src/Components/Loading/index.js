import React from "react";
import { Container, LoadingGif } from "./style";

function Loading() {
  return (
    <Container>
      <LoadingGif
        source={require("../../assets/gifs/pokeball_loading.gif")}
        size={"100%"}
      />
    </Container>
  );
}

export default Loading;
