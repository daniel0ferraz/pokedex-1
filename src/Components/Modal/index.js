import React from "react";
import { Container } from "./style";

function Modal(props) {
  const { visible, children } = props;
  return <Container visible={visible !== false}>{children}</Container>;
}

export default Modal;
