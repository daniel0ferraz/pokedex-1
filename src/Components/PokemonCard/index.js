import React from "react";
import { Container, PokemonImage } from "./style";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import PokemonInfo from "../PokemonInfo";

function PokemonCard(props) {
  const { typeColors, data } = props;
  const history = useHistory();
  const cardColor = typeColors[`${data.pokemon.types[0].type.name}`].card;
  return (
    <Container
      bg={cardColor}
      onClick={() => history.push(`/about/${data.pokemon.name}`)}
    >
      <PokemonInfo pokemon={data.pokemon} />
      <PokemonImage source={data.photo} size={"59px"} />
    </Container>
  );
}

const mapStateToProps = (state) => ({
  typeColors: state.pokedex.types,
});

export default connect(mapStateToProps, null)(PokemonCard);
