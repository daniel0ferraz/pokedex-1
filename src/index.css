@font-face {
  font-family: "SF Pro Display";
  src: local("SF Pro Display"),
    url(./assets/fonts/SFProDisplay-Regular.ttf) format("truetype");
}

* {
  background-color: transparent;
  box-sizing: border-box;
  border: none;
  margin: 0;
  padding: 0;
  outline: none;
}
body {
  font-family: SF Pro Display;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

button {
  cursor: pointer;
}
