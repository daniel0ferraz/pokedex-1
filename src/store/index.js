import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import { Reducers } from "../reducers";
import Saga from "../saga";
import { apiMiddleware } from "redux-api-middleware";
import { composeWithDevTools } from "redux-devtools-extension";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  Reducers,
  composeWithDevTools(applyMiddleware(sagaMiddleware, apiMiddleware))
);

sagaMiddleware.run(Saga);

export { store };
