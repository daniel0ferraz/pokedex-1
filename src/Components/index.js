import Image from "./Image";
import Search from "./Search";
import PokemonCard from "./PokemonCard";
import Loading from "./Loading";
import PokemonInfo from "./PokemonInfo";
import Type from "./Type";

export { Image, Search, PokemonCard, Loading, PokemonInfo, Type };
