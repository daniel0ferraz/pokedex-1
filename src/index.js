import React from "react";
import ReactDOM from "react-dom";
import "./index.css";
import { Provider } from "react-redux";
import { store } from "./store";
import * as serviceWorker from "./serviceWorker";
import Routes from "./routes";

ReactDOM.render(
  <React.StrictMode>
    <Provider store={store}>
      <Routes />
    </Provider>
  </React.StrictMode>,
  document.getElementById("root")
);

serviceWorker.unregister();
