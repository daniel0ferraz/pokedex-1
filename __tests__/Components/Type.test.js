// Link.react.test.js
import React from "react";
import { Provider } from "react-redux";
import { store } from "../../src/store";
import { Type } from "../../src/Components";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
const types = [
  "bug",
  "dark",
  "dragon",
  "electric",
  "fairy",
  "fighting",
  "fire",
  "flying",
  "ghost",
  "grass",
  "ground",
  "ice",
  "normal",
  "poison",
  "psychic",
  "rock",
  "steel",
  "water",
];

test("Rendering Types", () => {
  types.map((type) => {
    const component = mount(
      <Provider store={store}>
        <Type type={type} />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
