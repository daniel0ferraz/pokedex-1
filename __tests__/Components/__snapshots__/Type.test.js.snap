// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Types 1`] = `ReactWrapper {}`;

exports[`Rendering Types 2`] = `ReactWrapper {}`;

exports[`Rendering Types 3`] = `ReactWrapper {}`;

exports[`Rendering Types 4`] = `ReactWrapper {}`;

exports[`Rendering Types 5`] = `ReactWrapper {}`;

exports[`Rendering Types 6`] = `ReactWrapper {}`;

exports[`Rendering Types 7`] = `ReactWrapper {}`;

exports[`Rendering Types 8`] = `ReactWrapper {}`;

exports[`Rendering Types 9`] = `ReactWrapper {}`;

exports[`Rendering Types 10`] = `ReactWrapper {}`;

exports[`Rendering Types 11`] = `ReactWrapper {}`;

exports[`Rendering Types 12`] = `ReactWrapper {}`;

exports[`Rendering Types 13`] = `ReactWrapper {}`;

exports[`Rendering Types 14`] = `ReactWrapper {}`;

exports[`Rendering Types 15`] = `ReactWrapper {}`;

exports[`Rendering Types 16`] = `ReactWrapper {}`;

exports[`Rendering Types 17`] = `ReactWrapper {}`;

exports[`Rendering Types 18`] = `ReactWrapper {}`;
